[setup_state::03] Docs (#316)
software-mansion/protostar: website/docs/tutorials/guides/testing.md

You can find all [assert signatures here](https://github.com/software-mansion/protostar/blob/master/cairo/protostar/asserts.cairo).

## `__setup__`

> Often while writing tests you have some setup work that needs to happen before tests run. The hook `__setup__` can simplify and speed up your tests. Use `context` hint local to pass data from `__setup__` to test functions as demonstrated on the example below:

Reviewer: ```suggestion
Often while writing tests you have some setup work that needs to happen before tests run. The hook `__setup__` can simplify and speed up your tests. Use `context` variable to pass data from `__setup__` to test functions as demonstrated on the example below:
```

```cairo
%lang starknet

@view
func setup_state():

>     %{ tmp_state.contract_a_address = deploy_contract("./tests/integration/testing_hooks/basic_contract.cairo").contract_address %}

Author: `context`

    return ()
end

@view
func test_something():
    tempvar contract_address
    %{ ids.contract_address = context.contract_a_address %}

    # ...

    return ()
end
```
:::warning
Hint local `context` can only store strings, integers and booleans.
:::

:::info

> Protostar executes `__setup__` only once per a [test suite](https://en.wikipedia.org/wiki/Test_suite). Then, for each test case Protostar copies the StarkNet state and `context` object.

Reviewer: I think we don't need this one. We don't explain harder words ex. `StarkNet state`, and it's not our job. 
```suggestion
Protostar executes `__setup__` only once per a test suite. Then, for each test case Protostar copies the StarkNet state and `context` object.
```

Author: We don't require technical background from the target audience, and we use `test suite` term in the CLI output, so it's worth to at least add a link. It doesn't change anything for the guys familiar with this term.
`pros` > `cons`

```
Deploys a contract given a path relative to a Protostar project root. The section [Deploying contracts from tests](#deploying-contracts-from-tests) demonstrates a usage of this cheatcode.

:::warning

> Deploying contract is a slow operation. If it's possible try using this cheatcode in the [`__setup__` hook](#__setup__).

Reviewer: ```suggestion
Deploying a contract is a slow operation. If it's possible try using this cheatcode in the [`__setup__` hook](#__setup__).
```
